contracts/src/CrossChainApplications/NativeTokenBridge/NativeTokenDestination.sol

```diff
@@ -0,0 +1,188 @@
+//SPDX-License-Identifier: MIT
+pragma solidity 0.8.18;
+
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@subnet-evm-contracts/interfaces/IWarpMessenger.sol";
+import "@subnet-evm-contracts/interfaces/IAllowList.sol";
+import "@subnet-evm-contracts/interfaces/INativeMinter.sol";
+import "./INativeTokenDestination.sol";
+import "../../Teleporter/ITeleporterMessenger.sol";
+import "../../Teleporter/ITeleporterReceiver.sol";
+import "../../Teleporter/SafeERC20TransferFrom.sol";
+
+// Precompiled Native Minter Contract Address
+address constant MINTER_ADDRESS = 0x0200000000000000000000000000000000000001;
+// This is the address where the burned base fee go. We use this to send the amount of
+// burned tokens back to the source chain.
+address constant BURNED_TX_FEES_ADDRESS = 0x0100000000000000000000000000000000000000;
```

Comment:
Slightly unrelated to this PR, but what do we do with the burned base fees on the source chain? Do we then burn it on the source chain?

Reply:
I assume you mean "what do we do with the burned base fees on the source chain"? We will implement a mechanism for the destination chain to report the total value burned through fees to the source chain, allowing the source to burn the equivalent number of locked tokens there.